pkg/controller/utils/utils.go

```diff
@@ -161,3 +147,47 @@ const (
 func GetConcurrentReconciles() int {
 	return concurrentControllerReconciles
 }
+
+// MergeJsons will merge the global and local pull secret and return it
+func MergeJsons(globalPullSecret string, localPullSecret string, cdLog log.FieldLogger) (string, error) {
+
+	type dockerConfig map[string]interface{}
+	type dockerConfigJSON struct {
+		Auths       dockerConfig      `json:"auths"`
+		HTTPHeaders map[string]string `json:"HttpHeaders,omitempty"`
+	}
+
+	var mGlobal, mLocal dockerConfigJSON
+	jGlobal := []byte(globalPullSecret)
+	err := json.Unmarshal(jGlobal, &mGlobal)
+	if err != nil {
+		return "", err
+	}
+
+	jLocal := []byte(localPullSecret)
+	err = json.Unmarshal(jLocal, &mLocal)
+	if err != nil {
+		return "", err
+	}
+
+	for k, v := range mLocal.Auths {
+		if _, ok := mGlobal.Auths[k]; ok {
+			cdLog.Infof("The local pull secret is used for registry : %s instead of global pull secret", k)
```

Comment:
s/local/Cluster Deployment/

The term local is only meaningful within the code not to an end user that may see the log.